docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,659 @@
+```
+
+### Delete Issue
+
+Deletes a Jira issue and optionally its subtasks.
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                    |
+| ---------------------------- | --------- | -------- | ---------------------------------------------- |
+| `deleteIssue.issueKey`       | `string`  | Y        | The Jira Issue Key (e.g., EXT-123).            |
+| `deleteIssue.deleteSubtasks` | `boolean` | N        | If true, all subtasks will be deleted as well. |
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example deletes a Jira issue and all its subtasks when a promotion fails.
+
+```yaml
+steps:
+# existing steps create issue and other promotion steps
+# ....
+# ....
+# on failure cleanup logic
+- as: on-failure-cleanup-issue
+  uses: jira
+  if: ${{ failure() }}
+  config:
+    credentials:
+      secretName: jira-credentials
+    deleteIssue:
+      issueKey: "${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}"
+      deleteSubtasks: true
+```
+
+### Search Issues
+
+Searches for Jira issues using JQL ([Jira Query Language](https://support.atlassian.com/jira-service-management-cloud/docs/use-advanced-search-with-jira-query-language-jql/)).
+
+#### Configuration
+
+| Name                         | Type      | Required | Description                                                                                   |
+| ---------------------------- | --------- | -------- | --------------------------------------------------------------------------------------------- |
+| `searchIssue.jql`            | `string`  | Y        | The JQL query to search for issues.                                                           |
+| `searchIssue.expectMultiple` | `boolean` | N        | If true, expects multiple results and returns the first matching result. If false, expects single result and fails with >1 results. |
+| `searchIssue.fields`         | `array`   | N        | List of fields to include in search results.                                                  |
+| `searchIssue.expands`        | `array`   | N        | List of fields to expand in search results.                                                   |
+
+#### Output
+
+| Name    | Type     | Description                                                                 |
+| ------- | -------- | --------------------------------------------------------------------------- |
+| `issue` | `object` | The found Jira issue object containing all requested fields and expansions. |
+
+#### Example
+
+This example searches for open promotion issues in a specific project and expects multiple results.
+
+```yaml
+steps:
+- uses: jira
+  as: find-open-promotions
+  config:
+    credentials:
+      secretName: jira-credentials
+    searchIssue:
+      jql: 'project = DEPLOY AND status != "Done" AND labels = "${{ ctx.stage }}-promotion" AND created >= -7d'
```

Comment:
Feel free to change for consistency with other suggestions if you tweak them as well.

```suggestion
      jql: 'project = GUESTBOOK AND status != "Done" AND labels = "${{ ctx.stage }}-promotion" AND created >= -7d'
```